### PR TITLE
fix(ai): remove BUILD_ROUTER event when ai module is unloaded

### PR DESCRIPTION
**Pull request by soulbird**
### Description

Fixes # (issue)

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: t/plugin/ai3.t [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+            assert(apisix.handle_upstream == org_upstream, "disable: ai_upstream")
+            assert(apisix.http_balancer_phase == org_balancer_phase, "disable: http_balancer_phase")
+
+            -- enable ai plugin
+            local code = load_ai_module()
+            assert(code == 200, "enable ai plugin")
+            ngx.sleep(2)
+            local code = t('/hello', ngx.HTTP_GET)
+            assert(code == 200, "enable: access /hello")
+            -- TODO: It's not very reasonable, we need to fix it
```
Comment (spacewander) [inline t/plugin/ai3.t:147]: I am sorry I could not understand why it's not reasonable. Could you explain a little more in the comment?

**Review by spacewander — COMMENTED** [review]
Can we remove the TODO now?
https://github.com/apache/apisix/blob/48b84311067ae1b863cfcda23d079a262ff4a8ee/apisix/plugins/ai.lua#L299

File: t/plugin/ai3.t [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+                ngx.status = code
+                ngx.say(body)
+                return
+            end
+
+            local code = t('/hello', ngx.HTTP_GET)
+            assert(code == 200, "enable(default): access /hello")
+            assert(router.router_http.matching == ai_match, "enable(default): router_http.matching")
+            assert(apisix.handle_upstream == ai_upstream, "enable(default): ai_upstream")
+            assert(apisix.http_balancer_phase == ai_balancer_phase, "enable(default): http_balancer_phase")
```
Comment (membphis) [inline t/plugin/ai3.t:129]: it is not a good style, you are writing E2E test case
we can not use `unit test case` way

**Review by membphis — CHANGES_REQUESTED** [review]

**Review by spacewander — APPROVED** [review]

**Review by tzssangglass — APPROVED** [review]

**Comment by soulbird** [comment]
@membphis review pls

**Review by tzssangglass — APPROVED** [review]

**Review by membphis — APPROVED** [review]

**Comment by membphis** [comment]
@soulbird pls merge this PR soon ^_^

**Review by spacewander — APPROVED** [review]

**Pull request merged**